data/conf/rspamd/dynmaps/settings.php

```diff
@@ -31,19 +31,52 @@ function parse_email($email) {
 
 ?>
 settings {
-<?php
 
 /*
 // Start whitelist for forwarding hosts
 */
 
+<?php
+try {
+	$stmt = $pdo->query("SELECT `host` FROM `forwarding_hosts` WHERE `filter_spam` = 1");
+	$rows = $stmt->fetchAll(PDO::FETCH_COLUMN);
+}
+catch (PDOException $e) {
+	$rows = array();
+}
+
+if (!empty($rows)) {
+?>
+	whitelist_forwarding_hosts_with_spam_filter {
+		priority = high;
+<?php
+foreach ($rows as $host):
+?>
+		ip = "<?=$host;?>";
+<?php
+endforeach;
+?>

```

Comment:
I will merge it, but remove the "apply" directives. We should use multimaps with matching symbols. This will work by using honor_actions.